### PR TITLE
feat(Modal): create theme solution

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

**Comment by brandonlilly** [comment]
I made this PR to give a preview into how Styled System might integrate into our app, and as a forum to talking about more concrete details.

File: src/Modal/Modal.component.tsx [hunk 1]
```diff
@@ -18,73 +28,82 @@ import { ModalClose } from './Close';
...
-    margin: ${({ margin = 'auto' }: ModalProps) => margin};
+
+    ${compose(
+        marginStyles,
+        backgroundStyles,
+        colorStyles,
+        borderRadiusStyles
+    )}
+
+    ${size(
```
Comment (brandonlilly) [inline src/Modal/Modal.component.tsx:62]: This is using a custom `variant` util I made for handling more complex logic of theme settings.

The provided `variant` function from styled-components and styled-system would require each variant to be css style objects, which can't really handle this use-case.

Comment (brandonlilly) [reply src/Modal/Modal.component.tsx:62]: Alternatively I could do something like create a util for mapping the active variant properties as props to the component. This could potentially result in some prop name conflicts.

File: src/Modal/Header/Header.component.tsx [hunk 2]
```diff
@@ -15,16 +25,20 @@ export const StyledHeader = styled.div<ModalHeaderProps>`
...
-    color: ${({ color = colors.charcoal.light }) => color};
-    background: ${({ background = colors.white.base }) => background};
+    ${compose(
+        colorStyles,
+        backgroundStyles,
+        borderRadiusStyles
+    )}
+
+    && {
+        ${paddingStyles}
```
Comment (sharkparty) [inline src/Modal/Header/Header.component.tsx:40]: What's this doing?

Comment (brandonlilly) [reply src/Modal/Header/Header.component.tsx:40]: I separated out the padding like this because the padding is normally defined by the parent component (Modal). This  increases the specificity so that this css takes precedence if it exists.

Comment (sharkparty) [reply src/Modal/Header/Header.component.tsx:40]: Got it. Neat.

File: src/Modal/Modal.component.tsx [hunk 3]
```diff
@@ -18,73 +26,90 @@ import { ModalClose } from './Close';
...
+});
 
 const StyledModal = StyledReactModal.default.styled`
     position: relative;
-    width: ${({ width, size = defaultSize }: ModalProps) =>
-        width || `${Sizes[size].width}rem`};
-    height: ${({ height = '42.375rem' }: ModalProps) => height};
-    margin: ${({ margin = 'auto' }: ModalProps) => margin};
+
+    ${compose(
```
Comment (brandonlilly) [inline src/Modal/Modal.component.tsx:55]: This could be written alternatively like
```tsx
${marginStyles}
${backgroundStyles}
${colorStyles}
${borderRadiusStyles}
```

I wrote it with `compose` because Styled System recommends it for better performance of multiple styles https://styled-system.com/api#compose

File: src/Modal/Modal.component.tsx [hunk 4]
```diff
@@ -4,12 +4,20 @@ import * as React from 'react';
 // VENDOR
 import * as StyledReactModal from 'styled-react-modal';
 import styled, { css } from 'styled-components';
+import {
+    compose,
+    variant,
+    borderRadius as borderRadiusStyles,
```
Comment (brandonlilly) [inline src/Modal/Modal.component.tsx:10]: I aliased these to avoid shadow-naming tslint errors, due to the prop variables given defaults having the same names.

File: src/Modal/Modal.component.tsx (hunk 3 above)
Comment (sharkparty) [reply src/Modal/Modal.component.tsx:55]: Nah, I saw the docs on this and like the approach you took.

File: src/Modal/Modal.component.tsx (hunk 4 above)
Comment (sharkparty) [reply src/Modal/Modal.component.tsx:10]: I found this out when trying to delete them locally. Not a bad way to avoid the linter errors. We could modify the rules, but I like this better.

File: src/Modal/Content/Content.component.tsx [hunk 5]
```diff
@@ -9,12 +20,25 @@ export const StyledContent = styled.div<ModalContentProps>`
...
+        ${paddingStyles}
+    }
+
     order: 0;
     flex-grow: 1;
 `;
 
-export interface ModalContentProps {
+export interface ModalContentProps
+    extends ColorProps,
```
Comment (bjohnsonrmn) [inline src/Modal/Content/Content.component.tsx:37]: This took me a bit to wrap my head around, but I like it ✅

File: src/Modal/Footer/__snapshots__/Footer.spec.tsx.snap [hunk 6]
```diff
@@ -22,7 +23,6 @@ exports[`Component: Modal.Footer should render 1`] = `
   -ms-flex-order: 1;
   order: 1;
   justify-self: flex-end;
-  background: transparent;
```
Comment (bjohnsonrmn) [inline src/Modal/Footer/__snapshots__/Footer.spec.tsx.snap:25]: Was removing this intentional?

File: src/Modal/Modal.component.tsx [hunk 7]
```diff
@@ -4,87 +4,110 @@ import * as React from 'react';
...
 import { ModalFooter, StyledFooter } from './Footer';
 import { ModalClose } from './Close';
 
-export type ModalSize = 'lg' | 'sm';
-
-const Sizes = {
-    lg: { width: 50, contentPadding: 2.5, footerHeight: 6.5 },
-    sm: { width: 25, contentPadding: 2, footerHeight: 5.5 },
+export const MODAL_THEME_KEY = 'modal';
+export const MODAL_THEME = {
```
Comment (bjohnsonrmn) [inline src/Modal/Modal.component.tsx:28]: This theme object is helpful for understanding modal variations 👍

File: src/Modal/Modal.component.tsx [hunk 8]
```diff
@@ -4,87 +4,110 @@ import * as React from 'react';
...
+    )}
+
+    ${size(
+        ({ contentPadding = 0, footerHeight, width, height }) => css`
+            height: ${({ height: propHeight }: ModalProps) =>
+                propHeight || `${height}rem`};
+            width: ${({ width: propWidth }: ModalProps) =>
+                propWidth || `${width}rem`};
+
+            // The Modal Content component only gets its bottom padding if
```
Comment (bjohnsonrmn) [inline src/Modal/Modal.component.tsx:69]: I wonder if this complexity would be easier to manage at a higher level. Perhaps a separate component that is just the `ModalContent` that could be used in cases where we know we need the full padding. Or perhaps even another theme option that could offer different frill levels? Not sure if either would be easier, but just tossing it out there.

File: src/theme/system.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
+// ANCHOR
+import { radii } from './sizes.theme';
+
+// VENDOR
+import {
+    get,
+    system,
+    TLengthStyledSystem,
```
Comment (bjohnsonrmn) [inline src/theme/system.ts:8]: What is `TLengthStyledSystem`?

File: src/theme/system.ts [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    BackgroundProps,
+    padding,
+    PaddingProps,
+    margin,
+    MarginProps,
+    display,
+    DisplayProps,
+} from 'styled-system';
+
+export const variant = <T extends {}>({
```
Comment (bjohnsonrmn) [inline src/theme/system.ts:27]: Is this still vendor code all the way down to lines 43?

Comment (brandonlilly) [reply src/theme/system.ts:27]: Just the imports

File: src/theme/system.ts (hunk 9 above)
Comment (brandonlilly) [reply src/theme/system.ts:8]: Using it to define the Type Defs for BorderRadiusProps
